### PR TITLE
fix: add support for catalog:, npm:, and workspace: protocol

### LINKED ISSUE #71: 🐛 Bug: Support version protocols: npm:, workspace:, etc.
I use the syntax of the title so that I can have two different major version of one project as a dependency (one `@1.x.x` and one `2.x.x`), but this validator doesn't recognize this syntax and outputs an error instead. It would be cool if this syntax was supported, but given that it is quite uncommon (to the best of my knowledge) I don't think this should be a priority.

The following `package.json` content can be used to reproduce this error.

```json
{
  "name": "project",
  "version": "1.0.0",
  "private": true,
  "description": "...",
  "homepage": "https://github.com/ericcornelissen/project",
  "license": "MIT",
  "main": "lib/index.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/ericcornelissen/project"
  },
  "dependencies": {
    "svgo-v1": "npm:svgo@1.3.2",
    "svgo-v2": "npm:svgo@2.0.3"
  }
}
```

### PR DESCRIPTION
**Pull request by michaelfaith**
## PR Checklist

- [x] Addresses an existing open issue: fixes https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/509 and #71
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/package-json-validator/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/package-json-validator/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This change adds support for pnpm's catalog: protocol as well as yarn and pnpm's workspace protocol. Catalog entries can be simply `catalog:` or include a named catalog record (e.g. `catalog:react19`).  Workspace, can just be `workspace:` or workspace with a range specifier (`workspace:^`) or workspace and a specific version range (`workspace:^1.50`).

Workspace protocol: https://pnpm.io/next/workspaces#workspace-protocol-workspace
Catalog: https://pnpm.io/next/catalogs

Closes #71 (and possible https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/509)

### CODE REVIEW

File: PJV.test.mjs [hunk 1]
```diff
@@ -121,6 +121,15 @@ describe("NPM", () => {
...
 					"caret-top": "^1",
+					"workspace-package-no-range": "workspace:",
+					"workspace-package-caret": "workspace:^",
+					"workspace-package-any": "workspace:*",
+					"workspace-package-tilde-version": "workspace:~1.2.3",
+					"workspace-gt-version": "workspace:>1.2.3",
+					"catalog-package": "catalog:",
+					"catalog-named-package": "catalog:react19",
+					"svgo-v1": "npm:svgo@1.3.2",
+					"svgo-v2": "npm:svgo@2.0.3",
```
Comment (JoshuaKGoldberg) [inline PJV.test.mjs:133]: [Testing] These are nice and thorough tests for _valid_ cases. But could you also add some testing for _invalid_ too?

There are a few `test("...", function () { ... }`s after the `"smoke"` test. I think adding one of those would probably be good for testing specifically the protocols. WDYT?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Great start, thanks! 🎉

File: PJV.test.mjs (hunk 1 above)
Comment (michaelfaith) [reply PJV.test.mjs:133]: Done!

**Review by JoshuaKGoldberg — APPROVED** [review]
Great! 🚀

**Pull request merged**